### PR TITLE
Add Snap website field to Snap settings page and make UI changes

### LINKED ISSUE #20988: Link to allowlisted website field in Snap settings
Screenshot 2023-09-21 at 8 51 13 AM

In the above screenshot, a row is added for the website if a website is specified in the allowlist

Also, the install origin should not be a link and the connected sites far below this could be styled differently because they are not links

### PR DESCRIPTION
**Pull request by david0xd**
## **Description**
This PR adds a website URL to the Snap settings page and makes coupe of changes to the UI elements as requested in related task.

PR summary:
- Add new selector `getSnapRegistryData` for getting Snap registry data.
- Add Snap website URL to Snap Authorship component.
- Change origin to be plain text instead of link.
- Change color of text to default for connected websites.
- Check website against Phishing list using `PhishingController` before displaying it and display it only if it's not blacklisted.

## **Manual testing steps**
1. Install one or more Snaps.
2. Go to Settings page.
3. Click on one of the Snaps listed.
4. Check the UI changes requested in related task.

### **Before**
Screenshot 2023-09-28 at 16 11 29
Screenshot 2023-09-28 at 16 11 48

### **After**
Screenshot 2023-09-28 at 15 21 58
Screenshot 2023-09-28 at 16 07 43
Screenshot 2023-10-02 at 17 18 33

## **Related issues**
Fixes: https://github.com/MetaMask/metamask-extension/issues/20988

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/selectors/selectors.test.js [hunk 1]
```diff
@@ -780,4 +780,14 @@ describe('Selectors', () => {
     isInfuraBlocked = selectors.getInfuraBlocked(modifiedMockState);
     expect(isInfuraBlocked).toBe(true);
   });
+
+  it('#getSnapRegistryData', () => {
+    const mockSnapId = 'npm:@metamask/test-snap-bip44';
+    expect(selectors.getSnapRegistryData(mockState, mockSnapId)).toStrictEqual(
+      expect.objectContaining({
+        id: mockSnapId,
+        metadata: expect.anything(),
```
Comment (FrederikBolding) [inline ui/selectors/selectors.test.js:789]: Can we assert the metadata or the version?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js [hunk 2]
```diff
@@ -53,6 +58,25 @@ const SnapAuthorshipExpanded = ({ snapId, className, snap }) => {
   const subjectMetadata = useSelector((state) =>
     getTargetSubjectMetadata(state, snapId),
   );
+  const snapRegistryData = useSelector((state) =>
+    getSnapRegistryData(state, snapId),
+  );
+  const { website = undefined } = snapRegistryData
+    ? snapRegistryData.metadata
```
[the remark this answers is not in the document]
Comment (david0xd) [reply ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js]: Yes, added as suggested.

File: ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js [hunk 3]
```diff
@@ -53,6 +58,23 @@ const SnapAuthorshipExpanded = ({ snapId, className, snap }) => {
   const subjectMetadata = useSelector((state) =>
     getTargetSubjectMetadata(state, snapId),
   );
+  const snapRegistryData = useSelector((state) =>
+    getSnapRegistryData(state, snapId),
+  );
+  const { website = undefined } = snapRegistryData?.metadata ?? null;
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js:64]:
```suggestion
  const { website = undefined } = snapRegistryData?.metadata ?? {};
```
I don't think you can destructure `null`?

Comment (david0xd) [reply ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js]: True, my mistake. Updated it now.

File: ui/hooks/usePhishingController.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { PhishingController } from '@metamask/phishing-controller';
+
+/**
+ * React hook exposing the functionalities of PhishingController.
+ * This allows usage of PhishingController from components.
+ *
+ * @returns Object with checkPhishing function that accepts a website as an argument.
+ */
+export function usePhishingController() {
+  const [phishingController] = useState(new PhishingController());
```
[the remark this answers is not in the document]
Comment (david0xd) [reply ui/hooks/usePhishingController.js]: Done by @GuillaumeRx

File: app/scripts/metamask-controller.js [hunk 5]
```diff
@@ -2786,6 +2787,11 @@ export default class MetamaskController extends EventEmitter {
       dismissNotifications: this.dismissNotifications.bind(this),
       markNotificationsAsRead: this.markNotificationsAsRead.bind(this),
       updateCaveat: this.updateCaveat.bind(this),
+      testOrigin: async (origin) => {
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:2790]: Maybe this should be named something more descriptive?

Comment (david0xd) [reply app/scripts/metamask-controller.js]: Updated. Is it better? Any other suggestions?

File: ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js [hunk 6]
```diff
@@ -53,6 +61,22 @@ const SnapAuthorshipExpanded = ({ snapId, className, snap }) => {
...
+
+  useEffect(() => {
+    const performPhishingCheck = async () => {
+      const phishingResult = await getPhishingResult(website);
+
+      if (!phishingResult.result) {
+        setSafeWebsite(website);
+      }
+    };
+    performPhishingCheck();
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js:78]: We don't need to do this if the website is undefined.

File: ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js [hunk 7]
```diff
@@ -53,6 +61,24 @@ const SnapAuthorshipExpanded = ({ snapId, className, snap }) => {
   const subjectMetadata = useSelector((state) =>
     getTargetSubjectMetadata(state, snapId),
   );
+  const snapRegistryData = useSelector((state) =>
+    getSnapRegistryData(state, snapId),
+  );
+  const { website = undefined } = snapRegistryData?.metadata ?? {};
+  const [safeWebsite, setSafeWebsite] = useState('');
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js:68]: Should this be `null` or `undefined` by default?

File: ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js (hunk 6 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js]: Added check for website before calling.

File: ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js (hunk 7 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js]: Updated to use `null` now.

**Review by FrederikBolding — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**